torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -604,7 +604,12 @@ def _sharded_pre_load_state_dict_hook(
             fqn_from_global_root = f"{prefix}{FSDP_PREFIX}{fqn}"
         else:
             fqn_from_global_root = f"{prefix}{fqn}"
-        param = state_dict.pop(fqn_from_global_root)
+        try:
+            param = state_dict.pop(fqn_from_global_root)
+        except KeyError:
+            logger.warning(f"Did not find param with FQN {fqn_from_global_root}, skipping it. The weight will not be filled if you expect it to be.")
+            continue  # TODO: only don't crash for strict=False
```

Comment:
The TODO is not necessary. The error will be delayed but will be captured by nn.Module.load_state_dict.

Reply:
We can add a TODO here to adding UTs, https://github.com/pytorch/pytorch/issues/109134.